gensim/test/test_fasttext.py

```diff
@@ -1300,6 +1306,253 @@ def _run(self, fin):
+            self.assertEqual(model_trained.trainables.bucket, model_loaded.trainables.bucket)
+            self.assertEqual(model_trained.wv.min_n, model_loaded.wv.min_n)
+            self.assertEqual(model_trained.wv.max_n, model_loaded.wv.max_n)
+            self.assertEqual(model_trained.vocabulary.sample, model_loaded.vocabulary.sample)
+            self.assertEqual(set(model_trained.wv.index2word), set(model_loaded.wv.index2word))
+
+            for w in model_trained.wv.index2word:
+                v_orig = model_trained.wv[w]
+                v_loaded = model_loaded.wv[w]
+
+                try:
+                    self.assertLess(calc_max_diff(v_orig, v_loaded), MAX_WORDVEC_COMPONENT_DIFFERENCE)
+                except AssertionError as e:
+                    s = e.args[0]
+                    s += " (max difference for components in wordvector of \"%s\"" % w
+                    s += " larger than the thershold %f" % MAX_WORDVEC_COMPONENT_DIFFERENCE
+                    e.args = (s,)
+                    raise e
+
+    def test_round_trip_model_model_skipgram(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 1,
+            "negative": 5,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_model_model(model_params)
+
+    def test_round_trip_model_model_cbow(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 0,
+            "negative": 5,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_model_model(model_params)
+
+
+def _read_binary_file(fname):
+    with open(fname, "rb") as f:
+        data = f.read()
+    return data
+
+
+class SaveFacebookFormatFileGensimTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create binary fastText file model1.bin using gensim
+    + load file model1.bin to variable `model`
+    + save `model` to model2.bin
+    + check if files model1.bin and model2.bin are byte identical
+    """
+
+    def _check_roundtrip_file_file(self, model_params):
+        with temporary_file("roundtrip_file_to_file1.bin") as fpath1, \
+            temporary_file("roundtrip_file_to_file2.bin") as fpath2:
+            _create_and_save_test_model(fpath1, model_params)
+            model = gensim.models.fasttext.load_facebook_model(fpath1)
+            gensim.models.fasttext.save_facebook_model(model, fpath2)
+            self.assertEqual(_read_binary_file(fpath1), _read_binary_file(fpath2))
+
+    def test_roundtrip_file_file_skipgram(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 1,
+            "negative": 0,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_file_file(model_params)
```

Comment:
Unwrap model params here, as above, and everywhere else, please.

Reply:
Corrected everywhere.